diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 6e219b399..eadc7d454 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -338,7 +338,7 @@ config("compiler") {
       # links.
       ldflags += [ "-Wl,--build-id=sha1" ]
     } else if (current_os != "aix") {
-      ldflags += [ "-Wl,--build-id" ]
+      ldflags += [ "-Wl" ]
     }

     if (!is_android) {
diff --git a/toolchain/gcc_toolchain.gni b/toolchain/gcc_toolchain.gni
index 1c7bf7e7a..056c1070c 100644
--- a/toolchain/gcc_toolchain.gni
+++ b/toolchain/gcc_toolchain.gni
@@ -601,6 +601,7 @@ template("clang_toolchain") {
                            [
                              "strip",
                              "default_shlib_subdir",
+                             "extra_ldflags",
                              "enable_linker_map",
                              "use_unstripped_as_runtime_outputs",
                            ])
diff --git a/toolchain/linux/BUILD.gn b/toolchain/linux/BUILD.gn
index fa8b17e9d..de4d7572d 100644
--- a/toolchain/linux/BUILD.gn
+++ b/toolchain/linux/BUILD.gn
@@ -119,6 +119,8 @@ gcc_toolchain("x86") {
 }

 clang_toolchain("clang_x64") {
+  extra_ldflags = "-Wl,-rpath=\\\$ORIGIN -Wl,--enable-new-dtags"
+
   # Output linker map files for binary size analysis.
   enable_linker_map = true

